src/mongocxx/exception/error_category.hpp

```diff
@@ -10,29 +10,39 @@
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
-// limitations under the License./**
+// limitations under the License.
+
+#pragma once
 
 #include <mongocxx/config/prelude.hpp>
 
-#include <cstdint>
-#include <string>
+#include <system_error>
 
-// TODO: figure out if we need this
-// TODO: figure out if bson document should be view/value, etc.
+#include <mongocxx/exception/mongoc_error.hpp>
 
 namespace mongocxx {
 MONGOCXX_INLINE_NAMESPACE_BEGIN
-namespace error {
+namespace exception {
+
+const std::error_category& mongoc_error_category();
+
+std::error_code error_code_from_mongoc_error(mongoc_error error);
```

Comment:
Aha! This explains the confusion I had around why we needed these methods. Thanks!